spotify-api/spotifyRequests.ts

```diff
@@ -206,3 +206,213 @@ export const fetchArtistImagesBySpotifyId = async (
       throw err;
     });
 };
+
+export interface SpotifyYoutubeTrackMapping {
+  youtube: {
+    name: string;
+  };
+  spotify?: {
+    name: string;
+    uri: string;
+  };
+}
+const fetchSingleSpotifyTrackByYoutubePlaylistURL = (
+  accessToken: string,
+  playlistTitle: { title: string },
+  maxRetries: number = 3,
+  currentRetry: number = 0
+): Promise<SpotifyYoutubeTrackMapping> => {
+  return axios
+    .get<SpotifyApi.SearchResponse>(
+      queryParamsStringify('https://api.spotify.com/v1/search', {
+        q: playlistTitle.title,
+        type: 'track',
+        limit: '1',
+      }),
+      {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      }
+    )
+    .then((response) => {
+      // item limit is 1 above
+      const SpotifyTrackInfo: SpotifyYoutubeTrackMapping | undefined = response.data.tracks?.items.map((track) => {
+        return {
+          youtube: {
+            name: playlistTitle.title,
+          },
+          spotify: {
+            uri: track.uri,
+            name: track.name,
+          },
+        };
+      })[0];
+
+      if (SpotifyTrackInfo === undefined) {
+        throw new Error('No spotify track found for video with name ' + playlistTitle.title);
+      }
+
+      return SpotifyTrackInfo;
+    })
+    .catch((err: AxiosError) => {
+      if (err.response?.status === HttpStatusCode.TooManyRequests) {
+        if (currentRetry < maxRetries) {
+          // Retry the request after a delay
+          const retryDelayMs = parseInt((err.response!.headers['retry-after'] as string) ?? 10) * 1000;
+          return new Promise((resolve) => {
+            setTimeout(() => {
+              resolve(
+                fetchSingleSpotifyTrackByYoutubePlaylistURL(accessToken, playlistTitle, maxRetries, currentRetry + 1)
+              );
+            }, retryDelayMs);
+          });
+        } else {
+          // Max retries reached,
+          return {
+            youtube: {
+              name: playlistTitle.title,
+            },
+          };
+        }
+      } else {
+        // Can't find a corresponding track for the youtube video
+        return {
+          youtube: {
+            name: playlistTitle.title,
+          },
+        };
+      }
+    });
+};
+
+function partitionArray<T>(arr: T[], chunkSize: number) {
+  const result = [];
+  for (let i = 0; i < arr.length; i += chunkSize) {
+    result.push(arr.slice(i, i + chunkSize));
+  }
+  return result;
+}
```

Comment:
Put this function in `utils.ts`

Reply:
I don't think there is any benefit of this, since the only one file uses `partitionArrary`.